From 69458c734a481e46961a20f383356a363da73cf2 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Thu, 26 Nov 2009 08:49:17 +1000
Patch-mainline: 2.6.33
References: fate#310916
Git-commit: 54d9cb47dd6a754e434e5adeccb3a1e2835594fd
Subject: [PATCH 0131/2588] drm/radeon/kms/dp: fix return in dpcd retrival.

Not returning here caused us to get a display port version of 0 for everything
this caused power up to not get sent which ends up in a black screen.

Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit 54d9cb47dd6a754e434e5adeccb3a1e2835594fd)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/radeon/atombios_dp.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/radeon/atombios_dp.c b/drivers/gpu/drm/radeon/atombios_dp.c
index ebaf3f8..65c8239 100644
--- a/drivers/gpu/drm/radeon/atombios_dp.c
+++ b/drivers/gpu/drm/radeon/atombios_dp.c
@@ -459,6 +459,7 @@ void radeon_dp_getdpcd(struct radeon_connector *radeon_connector)
 				printk("%02x ", msg[i]);
 			printk("\n");
 		}
+		return;
 	}
 	dig_connector->dpcd[0] = 0;
 	return;
-- 
1.7.6

